crates/continuations/src/lib.rs

```diff
@@ -157,6 +157,36 @@ impl StackChain {
     pub fn is_main_stack(&self) -> bool {
         matches!(self, StackChain::MainStack(_))
     }
+
+    // We don't implement IntoIterator because our iterator is unsafe, so at
+    // least this gives us some way of indicating this, even though the actual
+    // unsafety lies in the `next` function.
+    pub unsafe fn into_iter(self) -> ContinuationChainIterator {
+        ContinuationChainIterator(self)
+    }
+}
+
+pub struct ContinuationChainIterator(StackChain);
+
+impl Iterator for ContinuationChainIterator {
+    type Item = (Option<*mut ContinuationObject>, *mut StackLimits);
+
+    fn next(&mut self) -> Option<Self::Item> {
+        match self.0 {
+            StackChain::Absent => None,
+            StackChain::MainStack(ms) => {
+                let next = (None, ms);
+                self.0 = StackChain::Absent;
+                Some(next)
+            }
+            StackChain::Continuation(ptr) => {
```

Comment:
We should be able to make this a `NonNull` pointer to get rid of the unsafe usage here, right?

Reply:
No, any of the methods that dereference a `NonNull` are still `unsafe`.

Reply:
What would the advantage of that be over the current version?

Reply:
We wouldn't have any of the conversion into references... May be negligible.